src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzerTests.cs

```diff
@@ -128,6 +128,86 @@ class Target
             await VerifyAnalyzerAsyncCs(source, tfmAndOption);
         }
 
+        [Fact]
+        public async Task ProjectWithPlatformNeutralAssemblyPropertyTest()
+        {
+            var source = @"
+using System.Collections.Generic;
+using System.Runtime.Versioning;
+
+[assembly:SupportedOSPlatform(""linux"")] 
+public class Test
+{
+    private string program;
+
+    [SupportedOSPlatform(""windows"")] // Can overwrite parent browser support as cross platform
+    public string WindowsOnlyProgram => program;
+
+    [SupportedOSPlatform(""windows"")]
+    [SupportedOSPlatform(""ios"")]
+    [SupportedOSPlatform(""linux"")]
+    public string WindowsIosLinuxOnlyProgram => program; // referencing internal field, should not warn
+
+    [SupportedOSPlatform(""android"")] // Can overwrite parent browser support as cross platform
+    [SupportedOSPlatform(""browser"")]
+    public string AndroidBrowserOnlyProgram => program; // referencing internal field, should not warn
+
+    [UnsupportedOSPlatform(""linux"")]
+    public string UnsupportedLinuxProgram => program;
+
+    void CrossPlatformCallSite()
+    {   
+        // should warn as WindowsOnlyProgram is windows only
+        var a = {|#0:WindowsOnlyProgram|};  // This call site is reachable on: 'linux'. 'Test.WindowsOnlyProgram' is only supported on: 'windows'.
+        a = {|#1:UnsupportedLinuxProgram|}; // This call site is reachable on: 'linux'. 'Test.UnsupportedLinuxProgram' is unsupported on: 'linux'.
+        a = WindowsIosLinuxOnlyProgram;
+        a = [|AndroidBrowserOnlyProgram|]; // This call site is reachable on: 'linux'. 'Test.AndroidBrowserOnlyProgram' is only supported on: 'android', 'browser'.
```

Comment:
Any reason why this diagnostic is not annotated with `#2`? I would expect either all diagnostics to have no `#` prefix or all of them to have it for consistency.

Reply:
It is because the diagnostic parameter has a list of platforms: "'android', 'browser'" and the order of the platforms are not consistent and cause a test failure

Reply:
> order of the platforms are not consistent

We always try to keep the analyzer diagnostic messages deterministic between builds. Can you order the platform names in the message?

Reply:
> Can you order the platform names in the message

I guess I need to update the [Enumerator of the SmallDictionary](https://github.com/dotnet/roslyn-analyzers/blob/5bd4ce968d457bcc42edf8838af79b937d69cbeb/src/Utilities/Compiler/SmallDictionary.cs#L739) for that, to loop through left branches first, but do not see an easy solution for that.

Reply:
I gave it try, updated the enumerator to always return elements in order so that the smallest elements come first, but the key hashcode populated by `StringComparer.GetHashCode(string)` is not consistent for each run, therefore the order of keys also not consistent. So I cannot keep the platform names in order efficiently, I could sort the list before diagnostics but not sure that overhead is worth it.

Reply:
You shouldn’t order this in the dictionary, just call the Order linq routine for the argument passed to the diagnostic create. We do this all the time, it should be fine to call Order on arguments passed to diagnostic.

Reply:
> You shouldn’t order this in the dictionary, just call the Order linq routine for the argument passed to the diagnostic create. 

Yes, as i mentioned in my previous comment i can call the `List.Sort()` method on the platforms list, my point is i do not really like to add such addition unless necessary, even it has a very minor impact.

>  We do this all the time, it should be fine to call Order on arguments passed to diagnostic.

After adding the sort but it is causing some tests fail where we are appending `and all other platforms` to build more meaningful message, of course i can tweak that somehow to fix it, but still i don't think it is worth. 

```log
Expected: This call site is reachable on: 'windows' 9.0 and before, and all other platforms. 'AllowList.Windows10Only()' is only supported on: 'windows' 10.0 and later.
Actual:   This call site is reachable on: and all other platforms, 'windows' 9.0 and before. 'AllowList.Windows10Only()' is only supported on: 'windows' 10.0 and later.
```